e2e-tests/common/assertions.py

```diff
@@ -1,8 +1,23 @@
 import json
+from typing import Dict, List, Optional
 
 
-def equal_json_strings(expected, actual):
-    return _ordered(json.loads(expected)) == _ordered(json.loads(actual))
+def equal_json_strings(
+    expected: str, actual: str, ignore_keys: Optional[List[str]] = None
+):
+    if not ignore_keys:
+        ignore_keys = []
+    return _equal_dicts(
+        _ordered(json.loads(expected)),
+        _ordered(json.loads(actual)),
+        ignore_keys=ignore_keys,
+    )
+
+
+def _equal_dicts(dict1: Dict, dict2: Dict, ignore_keys: List[str]):
```

Comment:
New method to compare dicts while ignoring specific keys (`recognizer_metadata` in our case, as the value `recognizer_metadata.recognizer_identifier` is not deterministic.